remote-api/src/main/java/org/alfresco/rest/api/impl/rules/NodeValidator.java

```diff
@@ -142,6 +142,24 @@ private NodeRef validateNode(final String nodeId, final QName expectedType, fina
         return nodeRef;
     }
 
+    private void validateChangePermission(boolean requireChangePermission, NodeRef nodeRef)
```

Comment:
Method name is confusing for me, because you could try to validate change permission without requirement for change permission: `validateChangePermission(false, nodeRef);` - in this case you are validating read permission. You could simply change method name to e.g. `validatePermission()`.

Reply:
Or you could solve it more generally and pass expected permission as an input param instead of boolean, e.g.:
`
validatePermission(NodeRef nodeRef, String expectedPermission) {
if (READ_PERMISSIONS.equals(expectedPermission) {
  if (permissionService.hasReadPermission(nodeRef) != ALLOWED)...
else
  if (permissionService.hasPermission(nodeRef, expectedPermission) != ALLOWED)...
`

Reply:
Solution is on you :)

Reply:
Thanks for the feedback Krystian. This method got created to avoid duplicate code, as you can see from the removed lines. I think that changing the boolean in the method would require a lot more work in a bigger scope as the argument gets passed from the methods that are calling this one. Do you think we could rename the method to something like checkPermissions to make it easier to understand and avoid changing the code?

Reply:
Of course, I'm ok with the method name update fix.